graphql_compiler/compiler/emit_sql.py

```diff
@@ -1,285 +1,110 @@
-    'query_path_to_selectable',
-    # 'query_path_to_location_info': Dict[Tuple[str, ...], LocationInfo], inverse mapping from
-    # each query_path to the LocationInfo located at that query_path
-    'query_path_to_location_info',
-    # 'query_path_to_output_fields': Dict[Tuple[str, ...], Dict[str, Tuple[str, type, bool]]]
-    # mapping from each query path to a mapping from field alias to the field name, type, and
-    # renamed status. This tuple is used to construct the query outputs, and track when a name
-    # changes due to collapsing into a CTE.
-    'query_path_to_output_fields',
-    # 'query_path_to_filters': Dict[Tuple[str, ...], List[Filter]], mapping from each query_path
-    # to the Filter blocks that apply to that query path
-    'query_path_to_filters',
-    # 'query_path_to_node': Dict[Tuple[str, ...], SqlNode], mapping from each
-    # query_path to the SqlNode located at that query_path.
-    'query_path_to_node',
-    # 'sql_schema_info': SQLAlchemySchemaInfo containing all relevant schema information
-    'sql_schema_info',
-))
-
-
-def emit_code_from_ir(sql_schema_info, sql_query_tree):
+import six
+import sqlalchemy
+
+from . import blocks
+
+
+class CompilationState(object):
+    """Mutable class used to keep track of state while emitting a sql query."""
+
+    def __init__(self, sql_schema_info, ir):
+        """Initialize a CompilationState, setting the current location at the root of the query."""
+        # Metadata
+        self._sql_schema_info = sql_schema_info
+        self._ir = ir
+
+        # Current query location state. Only mutable by calling _relocate.
+        self._location = None  # the current location in the query
+        self._classname = None  # the classname at the current location
+        self._alias = None  # a sqlalchemy table Alias at the current location
+        self._aliases = {}  # mapping marked query paths to table Aliases representing them
+        self._relocate(ir.query_metadata_table.root_location)
+
+        # The query being constructed as the IR is processed
+        self._from_clause = self._alias  # the main sqlalchemy Selectable
+        self._outputs = []  # sqlalchemy Columns labelled correctly for output
+        self._filters = []  # sqlalchemy Expressions to be used in the where clause
+
+    def _relocate(self, new_location):
+        """Move to a different location in the query, updating the _classname and _alias."""
+        self._location = new_location
+        new_location_type = self._ir.query_metadata_table.get_location_info(new_location).type
+        self._classname = new_location_type.name
+        if self._location.query_path in self._aliases:
+            self._alias = self._aliases[self._location.query_path]
+        else:
+            # mark_name, _ = self._location.get_location_name()
+            # XXX Use a consistent mark name with other backends, or stick with the
+            #     one generated by sqlalchemy?
```

Comment:
opinions? Our generated names look like `Animal__out_Animal_OfSpecies___1` (full path + gremlin revisit count), whereas the sqlalchemy-generated names look like `Animal_3` (table name + disambiguation number).

Reply:
No preference. Seems like using SQLAlchemy's default names might be less work? If so, let's stick with them.